src/chainparams.cpp

```diff
@@ -316,6 +316,24 @@ const CChainParams &Params() {
     return *pCurrentParams;
 }
 
+void CChainParams::UpdateBuriedDeploymentParameters(Consensus::BuriedDeploymentPos deployment, int64_t nStartHeight)
+{
+        if (deployment == Consensus::BIP34_HEIGHT_ACTIVE) {
+                consensus.BIP34Height = nStartHeight;
+        }
+        if (deployment == Consensus::BIP65_HEIGHT_ACTIVE) {
+                consensus.BIP65Height = nStartHeight;
+        }
+        if (deployment == Consensus::BIP66_HEIGHT_ACTIVE) {
+                consensus.BIP66Height = nStartHeight;
+        }
+}
+
+void UpdateRegtestBuriedDeploymentParameters(Consensus::BuriedDeploymentPos deployment, int64_t nStartHeight)
```

Comment:
More generic to take the the chainparams petname. Maybe assert that is equal to CBaseChainParams::REGTEST.